drivers/clock_control/clock_control_nrf2_lfclk.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_lfclk
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <nrfs_clock.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+#define NRF2_LFCLK_HAS_LFXO									\
+	(DT_INST_NODE_HAS_PROP(0, lfxo_clock) &&						\
+	 DT_NODE_HAS_STATUS(DT_INST_PROP(0, lfxo_clock), okay))
+
+#if NRF2_LFCLK_HAS_LFXO
+#define NRF2_LFCLK_MAX_ACCURACY DT_INST_PROP(0, lfxo_accuracy_ppm)
+#else
+#define NRF2_LFCLK_MAX_ACCURACY DT_INST_PROP(0, synth_accuracy_ppm)
+#endif
+
+#define NRFS_CLOCK_TIMEOUT K_MSEC(CONFIG_CLOCK_CONTROL_NRF2_NRFS_CLOCK_TIMEOUT_MS)
+
+static const struct clock_options {
+	uint16_t accuracy : 15;
+	uint16_t precision : 1;
+	nrfs_clock_src_t src;
+} clock_options[] = {
```

Comment:
It looks like this array is sorted by accuracy (from the worst being the highest ppm number). If that's a requirement for the algorithm, it would be good to note it somewhere, maybe in a comment?